### ISSUE TITLE
Filtrar posts por categoria

### ISSUE THREAD

**Issue by eduardomarqu-es** [MacMagazine/app-iOS#183]
Seria interessante podemos filtrar os posts de acordo com o assunto (reviews, tutoriais, rumores, etc.).

**Comment by eduardomarqu-es** [comment]
Exemplo 1 (app Jovem Nerd):

![photo_2019-07-01 13 08 13](https://user-images.githubusercontent.com/25755596/60450946-523adb00-9c01-11e9-98ef-6500e49be15d.jpeg)

Exemplo 2 (Mail):

![IMG_5827](https://user-images.githubusercontent.com/25755596/60451238-063c6600-9c02-11e9-8119-d8ddb8339071.jpeg)
![IMG_5828](https://user-images.githubusercontent.com/25755596/60451246-089ec000-9c02-11e9-8994-6c9e2e8579a1.png)

**Comment by rfischmann** [comment]
Uma 3ª opção é a área "Grupos", no canto superior esquerdo dos Contatos:

![Imagem PNG](https://user-images.githubusercontent.com/1125474/60451379-4996d480-9c02-11e9-851e-d096ca390e77.png)

**Comment by rfischmann** [comment]
Sugestão relacionada que recebemos:

> Na verdade não é um problema, mas sim uma sugestão. Não sei se é possível implementar essa sugestão, mas seria interessante criar escalas ou categorias  de assuntos para receber alertas. O conteúdo de vcs é excepcional e a produção é bem rica e farta. Então acabam sendo enviados diversos alertas durante o dia.  Com a opção por categorias, haveria um filtro por interesse, para quem assim o desejasse.

**Comment by rfischmann** [comment]
Mais um que mandou essa sugestão: https://d.pr/i/MwKIvl

**Comment by ailtonvivaz** [comment]
Um 4º exemplo é o modo como a Apple faz nos app TV e Fitness+

![image](https://user-images.githubusercontent.com/12598328/105643329-ffbdb100-5e6d-11eb-967f-6f97fa9e5f12.png)

![image](https://user-images.githubusercontent.com/12598328/105643254-773f1080-5e6d-11eb-90d4-99a8d1c5a3bb.png)

**Comment by cassio-rossi** [comment]
Tudo isso é um pouco mais complicado pois requer a criação de uma API no portal. Existe uma provisória mas que precisa de ajustes.

**Comment by maclacerda** [comment]
@cassio-rossi não seria possível realizar uma filtragem de forma "local" ?? uma vez que temos as `categorias` dentro dos Posts seria apenas uma estratégia de aplicar um `filter` para pegar as categorias e exibir para que o usuário selecione a que lhe interessa para exibição dos posts.

**Comment by cassio-rossi** [comment]
@maclacerda sim, todo o conteúdo está salvo no CoreData, incluindo as categorias e uma flag para Destaques.

**Comment by maclacerda** [comment]
![filter](https://user-images.githubusercontent.com/4759987/124807771-85bd4800-df34-11eb-973e-1e677c80a7f6.gif)

@eduardomarqu-es @rfischmann 

Ilustrando as sugestões que vocês adicionaram aqui, a filtragem teria como gatilho esse ícone na barra de navegação, inicialmente pensei em exibir uma tela ou uma `Action Sheet` para o usuário selecionar a categoria a qual o mesmo gostaria de filtrar. 

Para vocês isso estaria ok? Ou teriam alguma outra sugestão de comportamento?

O `Action Sheet` seria algo "semelhante" a este:

![2021-07-07 at 15 06](https://user-images.githubusercontent.com/4759987/124808265-16942380-df35-11eb-9b30-34fd77401433.png)

Qualquer dúvida estou a disposição.

**Comment by ailtonvivaz** [comment]
Uma outra sugestão poderia ser o `UIMenu`.

![IMG_575D7768A0CD-1](https://user-images.githubusercontent.com/12598328/124809496-8eaf1900-df36-11eb-962c-e04191d29592.jpeg)

**Comment by maclacerda** [comment]
> Uma outra sugestão poderia ser o `UIMenu`.
> 
> ![IMG_575D7768A0CD-1](https://user-images.githubusercontent.com/12598328/124809496-8eaf1900-df36-11eb-962c-e04191d29592.jpeg)

No caso vc estaria dizendo para exibição das categorias para o usuário selecionar correto?

**Comment by ailtonvivaz** [comment]
> 
> No caso vc estaria dizendo para exibição das categorias para o usuário selecionar correto?

Isso mesmo. No caso a mesma lógica que vc mostrou (o ícone que fica fill ou não), mas usando o `UIMenu` com a categoria selecionada. Ou "Tudo", algo assim.

**Comment by cassio-rossi** [comment]
Esse filtro não é tão simples como parece, não basta apenas pegar o que vier nas categorias, pois tem categoria "interna" por assim dizer e também podemos querer grupos de categoria como no site, por exemplo: Apple > iPhone > iPhone 12, iPhone 13, etc... Eu fiz uma API para pegar as categorias diretamente do Wordpress mas ainda não está 100%. Precisamos ver com mais calma.

**Comment by GilvanFernandes** [comment]
> Esse filtro não é tão simples como parece, não basta apenas pegar o que vier nas categorias, pois tem categoria "interna" por assim dizer e também podemos querer grupos de categoria como no site, por exemplo: Apple > iPhone > iPhone 12, iPhone 13, etc... Eu fiz uma API para pegar as categorias diretamente do Wordpress mas ainda não está 100%. precisamos ver com mais calma.

E aí, Cassio. Ainda há algo para desenvolver no backend do site, como a funcionalidade API? Se precisarem, posso colaborar com essa tarefa.

**Comment by cassio-rossi** [comment]
> > Esse filtro não é tão simples como parece, não basta apenas pegar o que vier nas categorias, pois tem categoria "interna" por assim dizer e também podemos querer grupos de categoria como no site, por exemplo: Apple > iPhone > iPhone 12, iPhone 13, etc... Eu fiz uma API para pegar as categorias diretamente do Wordpress mas ainda não está 100%. precisamos ver com mais calma.
> 
> E aí, Cassio. Ainda há algo para desenvolver no backend do site, como a funcionalidade API? Se precisarem, posso colaborar com essa tarefa.

Vamos continuar conversando via Telegram. Obrigado.
